cns/service.go

```diff
@@ -226,8 +232,13 @@ func getTLSConfigFromKeyVault(tlsSettings localtls.TlsSettings, errChan chan<- e
 		errChan <- cr.Refresh(ctx, tlsSettings.KeyVaultCertificateRefreshInterval)
 	}()
 
+	minTLSVersionNumber, err := TLSVersionNumber(tlsSettings.MinTLSVersion)
+	if err != nil {
+		return nil, errors.Wrap(err, "MinTLSVersion is not valid")
```

Comment:
Usually, for error wrapping, I put what I was trying to do when the error emerged. Here it would be:

```suggestion
		return nil, errors.Wrap(err, "parsing MinTLSVersion from config")
```

Doing this gives you a progressive chain of explanation.